src/io/dirops.c

```diff
@@ -35,31 +35,28 @@ static int mkdir_p(MVMThreadContext *tc, wchar_t *pathname, MVMint64 mode) {
 #else
 static int mkdir_p(MVMThreadContext *tc, char *pathname, MVMint64 mode) {
     char *p = pathname, ch;
-    uv_fs_t req;
 #endif
+
     int created = 0;
 
-    for (;; ++p)
-        if (!*p || IS_SLASH(*p)) {
+    while (*p++)
+        if (*p == '\0' || IS_SLASH(*p)) {
```

Comment:
I can't understand here, if `*p == '\0'` is true, why we need `*p = '\0';` after `ch = *p;` ?

Reply:
I think other codes that's not about the bug should not be changed. Because it may causes more bugs. :)

Reply:
I changed the condition into comparing with `\0` to make it clear what we are looking for either end of string or a slash to call mkdir on (micro-optimization speaking I should change it into `IS_SLASH(*p) || *p == '\0'`).  If I understand the original code correctly, `ch` is a temporary character to store the current one when it is changed into the null terminator so that the mkdir call will be on the parent directory only.

Reply:
> I think other codes that's not about the bug should not be changed. Because it may causes more bugs. :)

I won't argue with that, reverting irrelevant changes.

Reply:
Well, could do make a long dir `/tmp/path/to/long/dir` in the gdb and print `p`  to understand it. I did it long time ago :)